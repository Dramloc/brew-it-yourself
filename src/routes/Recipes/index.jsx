import { Component } from 'preact';
import Helmet from 'preact-helmet';
import fetch from 'isomorphic-fetch';
import { Page } from '../../components/Page';

function RecipeList({ recipes }) {
	return <span>RecipeList</span>;
}

export default class Recipes extends Component {
	state = {
		recipes: null
	};

	fetchRecipes() {
		return fetch('https://api.punkapi.com/v2/beers')
			.then(response => response.json())
			.then(recipes => this.setState({ recipes }));
	}

	componentDidMount() {
		this.fetchRecipes();
	}

	render(_, { recipes }) {
		return (
			<Page>
				<Helmet title="Recipes" />
				<RecipeList recipes={recipes} />
			</Page>
		);
	}
}
