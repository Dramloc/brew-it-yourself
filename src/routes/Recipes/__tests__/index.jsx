import Helmet from 'preact-helmet';
import render from 'preact-render-spy';
import Recipes from '../index';

describe('<Recipes />', () => {
	test('<Recipes /> matches snapshot', () => {
		expect(render(<Recipes />)).toMatchSnapshot();
		expect(Helmet.peek()).toMatchSnapshot();
	});
});
