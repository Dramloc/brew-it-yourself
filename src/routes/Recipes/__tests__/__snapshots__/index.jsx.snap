// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recipes /> <Recipes /> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<main class="Page">
  <span>RecipeList</span>
</main>

`;

exports[`<Recipes /> <Recipes /> matches snapshot 2`] = `
Object {
  "baseTag": Array [],
  "htmlAttributes": Object {},
  "linkTags": Array [],
  "metaTags": Array [],
  "noscriptTags": Array [],
  "onChangeClientState": [Function],
  "scriptTags": Array [],
  "styleTags": Array [],
  "title": "Recipes",
  "titleAttributes": Object {},
}
`;
